jcl/src/java.base/share/classes/openj9/internal/criu/security/CRIUConfigurator.java

```diff
@@ -101,6 +152,52 @@ public static void setCRIURestoreMode() {
 					System.out.println(provider + " restored.");
 				}
 			}
+			// If any algorithms are available from when we took a checkpoint in our algorithm cache
+			// we can now allow callers to get values from this cache.
+			if (cachedAlgorithms.size() > 0) {
+				if (debug) {
+					System.out.println("Algorithm cache enabled.");
+				}
+			    isCacheEligible = true;
+			}
+		}
+	}
+
+	/**
+	 * Gets algorithms from the checkpointed cache of algorithms.
+	 *
+	 * @param serviceName
+	 */
+	public static Set<String> getAlgorithms(String serviceName) {
+		if ((serviceName == null) || (serviceName.isEmpty())) {
+			return Collections.emptySet();
 		}
+
+		if (debug) {
+			System.out.println("Returned cached algorithms for service <" + serviceName + ">: " + cachedAlgorithms.get(serviceName));
+		}
+		return cachedAlgorithms.get(serviceName);
+	}
+
+	/**
+	 * Checks if the algorithm cache populated during a checkpoint is available. This cache is
+	 * available when there is data in the cache and we have completed a CRIU restore. The cache
+	 * may also become invalid when any security providers were inserted or removed from the provider list
+	 * as dictated by the logic in the Security class.
+	 * @return boolean value true when the cache is available, false otherwise.
+	 */
+	public static boolean isCachedAlgorithmsPresentAndReady() {
+		return (cachedAlgorithms.size() > 0) && isCacheEligible;
```

Comment:
Please reverse this (do the least expensive test first):
```java
		return isCacheEligible && !cachedAlgorithms.isEmpty();
```

Reply:
Made this update as suggested.